src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -21,31 +23,79 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
     /// there won't be multiple threads calling this client at the same time.
     /// </remarks>
     internal class ConfigurationClientManager : IConfigurationClientManager
-    {
-        private readonly IList<ConfigurationClientWrapper> _clients;
-
-        public ConfigurationClientManager(IEnumerable<string> connectionStrings, ConfigurationClientOptions clientOptions)
-        {
-            if (connectionStrings == null || !connectionStrings.Any())
-            {
-                throw new ArgumentNullException(nameof(connectionStrings));
-            }
-
-            _clients = connectionStrings.Select(connectionString => 
-            {
-                var endpoint = new Uri(ConnectionStringParser.Parse(connectionString, ConnectionStringParser.EndpointSection));
-                return new ConfigurationClientWrapper(endpoint, new ConfigurationClient(connectionString, clientOptions));
-            }).ToList();
-        }
-
-        public ConfigurationClientManager(IEnumerable<Uri> endpoints, TokenCredential credential, ConfigurationClientOptions clientOptions)
-        {
-            if (endpoints == null || !endpoints.Any())
-            {
-                throw new ArgumentNullException(nameof(endpoints));
-            }
-
-            _clients = endpoints.Select(endpoint => new ConfigurationClientWrapper(endpoint, new ConfigurationClient(endpoint, credential, clientOptions))).ToList();
+    {
+        private IList<ConfigurationClientWrapper> _clients;
+
+        private IEnumerable<string> _connectionStrings;
+        private IEnumerable<Uri> _endpoints;
+        private TokenCredential _credential;
+        private ConfigurationClientOptions _clientOptions;
+        private Lazy<SrvLookupClient> _srvLookupClient;
+        private DateTimeOffset _lastFallbackClientRefresh = default;
+        private bool _allClientFailed = false;
+
+        private readonly Uri _endpoint;
+        private readonly string _secret;
+        private readonly string _id;
+        private readonly bool _isUsingConnetionString;

```

Comment:
Can't this be deduced by whether `credential` is null?

Reply:
Sounds not bad